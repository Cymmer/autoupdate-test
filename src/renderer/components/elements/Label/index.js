import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';

const Label = ({ children, htmlFor, disabled, className, id }) => (
  <label
    className={cn(className, {
      [styles.Label]: !disabled,
      [styles.Label___disabled]: disabled,
    })}
    htmlFor={htmlFor}
    id={id}
    type="radio"
  >
    {children}
  </label>
);

Label.defaultProps = {
  disabled: false,
  className: null,
  id: null,
};

Label.propTypes = {
  children: PropTypes.node.isRequired,
  htmlFor: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  className: PropTypes.string,
  id: PropTypes.string,
};

export default Label;
