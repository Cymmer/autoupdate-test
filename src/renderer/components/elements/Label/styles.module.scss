@use '../../../scss' as *;

@mixin base {
  @include body(0.875rem, 1.25);
  @include theme {
    color: t('textColor-700');
  }

  @content;
}

.Label {
  @include base;

  &___disabled {
    @include base {
      opacity: 0.5;
    }
  }
}
