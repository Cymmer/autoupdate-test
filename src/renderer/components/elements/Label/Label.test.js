import React from 'react';
import { shallow } from 'enzyme';
import Label from './index';

describe('RENDERING - Label', () => {
  it('should render with default props', () => {
    const component = shallow(<Label htmlFor="test">Test</Label>);
    expect(component).toMatchSnapshot();
  });

  it('should render the appropriate disabled class if disabled', () => {
    const component = shallow(
      <Label disabled htmlFor="test">
        Test
      </Label>
    );
    expect(component).toMatchSnapshot();
  });
});
