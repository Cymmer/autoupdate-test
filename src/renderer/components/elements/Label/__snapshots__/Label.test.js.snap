// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Label should render the appropriate disabled class if disabled 1`] = `
<label
  className="Label___disabled"
  htmlFor="test"
  id={null}
  type="radio"
>
  Test
</label>
`;

exports[`RENDERING - Label should render with default props 1`] = `
<label
  className="Label"
  htmlFor="test"
  id={null}
  type="radio"
>
  Test
</label>
`;
